roofit/roofitcore/src/RooAbsPdf.cxx

```diff
@@ -2289,7 +2289,17 @@ RooDataSet *RooAbsPdf::generate(const RooArgSet &whatVars, const RooDataSet& pro
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+
+std::string RooAbsPdf::translate(RooFit::Detail::CodeSquashContext &ctxManager) const
+{
+   // For now just return function/normalization integral.
+   std::string func = ctxManager.getResultName(servers()[0]->GetName());
+   std::string normIntg = ctxManager.getResultName(servers()[1]->GetName());
```

Comment:
```suggestion
   std::string const&  func = ctxManager.getResultName(servers()[0]);
   std::string const& normIntg = ctxManager.getResultName(servers()[1]);
```
Avoid string copying.

Also, I still don't understand why the `getResult()` function has to take the name directly. I have suggested this in one of the previous comments that didn't get fully addressed.

I think it's better to take `RooAbsArg const*` and then get the name in the implementation of the context. This has two advantages:
1. If we change the way that the results are looked up (which we probably will do at some point because string lookups are relatively expensive), we only have to change one line of code.
2. less verbose code in the translate functions

Was it just an oversight that you didn't address this, or a conscious decision because you have some arguments against it?

Reply:
I did this because from the last conversation I had with you about the keys of the map, you seemed fairly certain that you wanted the keys to be the names and not the name pointers (in favour of human readability). Considering the name string (or pointer) is the only way to ID a RooFit class, I kept it that way.

Reply:
I'm very sorry, there was a misunderstanding then! I thought we were talking about what we use a variable names in the code to store temporary results, where I advocated for the name and against using the name pointer address.

For the key in the map inside the squash context, the name pointer would be very appropriate (we do the same in the batch mode) too.